### ISSUE TITLE
Middleware issue

### ISSUE THREAD

**Issue by DavertMik** [Naktibalda/codeception-zend-expressive-tests#7]
My PR didn't fix the issue. The current one is:

```
PHP Fatal error:  Uncaught TypeError: Argument 1 passed to Zend\Expressive\Router\Middleware\RouteMiddleware::__construct() must implement interface Zend\Expressive\Router\RouterInterface, none given, called in /home/travis/build/Codeception/Codeception/frameworks-zend-expressive/vendor/zendframework/zend-expressive/src/MarshalMiddlewareTrait.php on line 159 and defined in /home/travis/build/Codeception/Codeception/frameworks-zend-expressive/vendor/zendframework/zend-expressive-router/src/Middleware/RouteMiddleware.php:51
Stack trace:
#0 /home/travis/build/Codeception/Codeception/frameworks-zend-expressive/vendor/zendframework/zend-expressive/src/MarshalMiddlewareTrait.php(159): Zend\Expressive\Router\Middleware\RouteMiddleware->__construct()
#1 /home/travis/build/Codeception/Codeception/frameworks-zend-expressive/vendor/zendframework/zend-expressive/src/MarshalMiddlewareTrait.php(94): Zend\Expressive\Application->marshalInvokableMiddleware('Zend\\Expressive...', Object(Zend\Diactoros\Response))
#2 /home/travis/build/Codecepti in /home/travis/build/Codeception/Codeception/frameworks-zend-expressive/vendor/zendframework/zend-expressive-router/src/Middleware/RouteMiddleware.php on line 51

```

https://travis-ci.org/Codeception/Codeception/jobs/352600641#L3974

**Comment by Naktibalda** [comment]
I looked around and stumbled upon this migration tool commit: https://github.com/zendframework/zend-expressive-tooling/commit/aa0dc78e3a8e79c88f2a8fc7095004fac355382c
```
This command does the following:

- Adds Zend\Expressive\Router\ConfigProvider to config/config.php
- Adds Zend\Expressive\ConfigProvider to config/config.php
- Replaces `pipeRoutingMiddleware()` calls with `pipe(\Zend\Expressive\Router\Middleware\RouteMiddleware::class)`
- Replaces `pipeDispatchMiddleware()` calls with `pipe(\Zend\Expressive\Router\Middleware\DispatchMiddleware::class)`
- Replaces `pipe()` calls that pipe `Implicit*Middleware` to reference zend-expressive-router variants
```

`- Adds Zend\Expressive\Router\ConfigProvider to config/config.php` looks relevant.
config/config.php hasn't been updated since initial commit, so I copied entire content of expected result from the test (yay, copy-paste driven development) and committed it to 2.2 branch https://github.com/Naktibalda/codeception-zend-expressive-tests/commit/48a14ed7c7be2adf0bcedf1ace16bc5a6af27891

**Comment by Naktibalda** [comment]
And I added zend-config-aggregator to composer.json

**Comment by Naktibalda** [comment]
The last error I got was:
```
[Zend\ServiceManager\Exception\InvalidServiceNameException]                  
  A service by the name or alias "Zend\Expressive\Router\Middleware\DispatchM  
  iddleware" already exists and cannot be overridden; please use an alternate  
   name
```

**Comment by DavertMik** [comment]
Maybe it doesn't need to be piped then?

**Comment by Naktibalda** [comment]
Fixed by #9
